New: Part I - Basic Features

quangvnguyen opened the pull request:
Complete Part I - Basic Features in on boarding project:

- Login feature

- Add New Note feature

quangvnguyen committed:
add Login feature

quangvnguyen committed:
add Add New Note feature

quangvnguyen committed:
Add config for eslint

leo-le-07 commented on src/screens/Login/index.tsx:
```diff
@@ -0,0 +1,95 @@
+  constructor(props) {
+    super(props);
+
+    this.state = {
```
We don't need to initialize state inside `constructor` method. It could be declared as class property such as:
```
class Login extends ... {
   state = {
   }
   ...
}
```

leo-le-07 commented on src/screens/Login/index.tsx:
```diff
@@ -0,0 +1,95 @@
+
+  validateForm() {
+    const { email, password } = this.state;
+    return (
```
```
return email.length > 0 && password.length > 0
```
still be short and meaningful, right?

leo-le-07 commented on src/screens/Login/index.tsx:
```diff
@@ -0,0 +1,95 @@
+          <FormGroup controlId="loginEmail" bsSize="large">
+            <ControlLabel>Email</ControlLabel>
+            <FormControl
+            autoFocus
```
Indentation please

leo-le-07 commented on src/screens/Login/index.tsx:
```diff
@@ -0,0 +1,95 @@
+          <FormGroup controlId="loginPassword" bsSize="large">
+            <ControlLabel>Password</ControlLabel>
+            <FormControl
+            value={password}
```
Indentation please

leo-le-07 commented on src/screens/Login/types.tsx:
```diff
@@ -0,0 +1,9 @@
+export interface LoginState {
```
Normally we'll name interface with prefix `I` as convention: `ILoginState` for example.

leo-le-07 commented on src/screens/Login/types.tsx:
```diff
@@ -0,0 +1,9 @@
+  isLoading: boolean;
+}
+
+export interface LoginProps {
```
Same here `ILoginProps`. Please apply this convention for the rest of your typescript definition's name.

leo-le-07 commented on src/screens/NoteNew/index.tsx:
```diff
@@ -0,0 +1,128 @@
+    return content.length > 0;
+  }
+
+  formatFilename(str) {
```
Humm what's the difference:
```
formatFilename(str) {
...
}
```
and
```
formatFilename = (str) => {
...
}
```
Please investigate it.

quangvnguyen replied:
As I know, the first function is function declaration, the second is arrow function. In this case, both functions are available because we don't use `this` in their scope. But I will change this to arrow function (the second one) for consistent. Thanks!

leo-le-07 replied:
That's great. I mean we should follow consistency for project we're working.

quangvnguyen committed:
Refactor: resolve issues in code review

- Change interface name as convention
- Change Id for not duplicated
- Change functions type as consistent
